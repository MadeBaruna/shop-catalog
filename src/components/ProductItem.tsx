import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Icon, Button } from 'semantic-ui-react';

const Div = styled.div`
  display: flex;
  flex-direction: column;
  background: white;
`;

const DetailContainer = styled.div`
  display: flex;
  flex-direction: row;
  padding: 8px 16px;
  margin-bottom: 8px;
`;

const Detail = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  min-width: 0; 

  p {
    font-size: 16px;
    margin-bottom: 4px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    margin-right: 8px;
  }

  span {
    background: #e8e8e8;
    align-self: flex-start;
    padding: 3px 8px;
  }

  h3 {
    margin-top: 8px;
  }
`;

const ButtonContainer = styled.div`
  margin-right: -8px;
`;

const BuyButton = styled(Button)`
  margin-left: 8px !important;
`;

interface IProps {
  id: string;
  name: string;
  pictureUrl: string;
  size: string;
  price: number;
}

const priceFormat = new Intl.NumberFormat('id-ID');

const ProductItem: React.FC<IProps> = ({
  id,
  name,
  size,
  price,
  pictureUrl
}) => (
  <Link to={`/product/${id}`}>
    <Div>
      <img src={pictureUrl} alt="gambar produk" />
      <DetailContainer>
        <Detail>
          <p>{name}</p>
          <span>{size}</span>
          <h3>{priceFormat.format(price)}</h3>
        </Detail>
        <ButtonContainer>
          <Icon name="heart outline" color="grey" size="large" />
          <BuyButton color="red">BELI</BuyButton>
        </ButtonContainer>
      </DetailContainer>
    </Div>
  </Link>
);

export default ProductItem;
