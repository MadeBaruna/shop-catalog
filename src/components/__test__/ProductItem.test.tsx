import React from 'react';

import ProductItem from '../ProductItem';
import { shallow } from 'enzyme';

it('renders <ProductItem />', () => {
  const wrapper = shallow(
    <ProductItem
      id="asd"
      name="This is a rather long product name"
      size="M, S, XL"
      pictureUrl="https://image.jpg"
      price={120000}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
