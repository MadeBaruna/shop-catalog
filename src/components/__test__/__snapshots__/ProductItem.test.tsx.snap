// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <ProductItem /> 1`] = `
<Link
  to="/product/asd"
>
  <styled.div>
    <img
      alt="gambar produk"
      src="https://image.jpg"
    />
    <styled.div>
      <styled.div>
        <p>
          This is a rather long product name
        </p>
        <span>
          M, S, XL
        </span>
        <h3>
          120,000
        </h3>
      </styled.div>
      <styled.div>
        <Icon
          as="i"
          color="grey"
          name="heart outline"
          size="large"
        />
        <Styled(Button)
          color="red"
        >
          BELI
        </Styled(Button)>
      </styled.div>
    </styled.div>
  </styled.div>
</Link>
`;
